### PR TITLE
MINOR: Fix toString method of IsolationLevel

### PR DESCRIPTION
**Pull request by PzaThief**
IsolationLevel exposed wild toString method, So uncomfortable toLowerCase is everywhere when we use toString. there was no other specific purpose to use toLowerCase and seems to using it is just for works.
So, I fixed it like OffsetResetStrategy's toString to do not expose wild method.

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by mjsax — APPROVED** [review]
Thanks for the PR. LGTM.

Retriggered Jenkins.

**Comment by mjsax** [comment]
Build failed again -- can you maybe rebase this PR on `trunk` to ensure it's on the latest version.

**Comment by PzaThief** [comment]
> Build failed again -- can you maybe rebase this PR on `trunk` to ensure it's on the latest version.

Sorry for missed it. Can you please retrigger it?

File: connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStore.java [hunk 1]
```diff
@@ -248,7 +247,7 @@ public void start() {
                         + "support for source connectors, or upgrade to a newer Kafka broker version.";
             } else {
                 message = "When " + ConsumerConfig.ISOLATION_LEVEL_CONFIG + "is set to "
-                        + IsolationLevel.READ_COMMITTED.name().toLowerCase(Locale.ROOT)
+                        + IsolationLevel.READ_COMMITTED
```
Comment (ashwinpankaj) [inline connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStore.java]:
```suggestion
                        + IsolationLevel.READ_COMMITTED.toString()
```

File: connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java [hunk 2]
```diff
@@ -234,7 +233,7 @@ public void start() {
             throw new ConnectException(
                     "Must provide a TopicAdmin to KafkaBasedLog when consumer is configured with "
                             + ConsumerConfig.ISOLATION_LEVEL_CONFIG + " set to "
-                            + IsolationLevel.READ_COMMITTED.name().toLowerCase(Locale.ROOT)
+                            + IsolationLevel.READ_COMMITTED
```
Comment (ashwinpankaj) [inline connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java]:
```suggestion
                            + IsolationLevel.READ_COMMITTED.toString()
```

File: connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStore.java (hunk 1 above)
Comment (PzaThief) [reply connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStore.java]: Oh my... I missed it.
Thank you for careful review!

**Comment by mimaison** [comment]
Re-kicked a build: https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14782/9/

**Comment by mjsax** [comment]
Thanks for the PR. Merged to `trunk`.

**Pull request merged**